cmd/tsidp/tsidp.go

```diff
@@ -611,6 +758,124 @@ func (s *idpServer) serveOpenIDConfig(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// funnelClient represents an OIDC client/relying party that is accessing the
+// IDP over Funnel.
+type funnelClient struct {
+	ID     string `json:"client_id"`
+	Secret string `json:"client_secret,omitempty"`
+	Name   string `json:"name,omitempty"`
+}
+
+// /clients is a privileged endpoint that allows the visitor to create new
+// Funnel-capable OIDC clients, so it is only accessible over the tailnet.
+func (s *idpServer) serveClients(w http.ResponseWriter, r *http.Request) {
+	if isFunnelRequest(r) {
+		http.Error(w, "tsidp: not found", http.StatusNotFound)
+		return
+	}
+
+	path := strings.TrimPrefix(r.URL.Path, "/clients/")
+
+	if path == "new" {
+		s.serveNewClient(w, r)
```

Comment:
this means that its open to all users in the tailnet which might not be desirable, probably fine in prototype phase but would need to change to something with ACLs

Reply:
Yep, I'll make a note of that